pkgs/by-name/au/auto-patchelf/source/auto-patchelf.py

```diff
@@ -394,14 +413,21 @@ def main() -> None:
         action="store_true",
         help="Attempt to search for and relink libc dependencies.",
     )
+    parser.add_argument(
+        "--no-add-existing",
```

Comment:
I think the name does not convey the right meaning. I would prefer something like `--no-default-paths` or `--no-self-references`.

The description below can be confusing to the readers too. The thing that is avoided here is adding the paths to patch to the search paths. Ir makes sense as you are supposed to prefer binaries that you are providing over system ones.

That is why I am a bit uncertain about this whole change. Until now this problem has always been solved by removing the libraries that autopatchelf should not consider, or by passing files in paths_to_patch.

The patch is well written, and there is no problem with it _per se_, but I would like to understand why and if it is really needed.

Reply:
> The patch is well written, and there is no problem with it *per se*, but I would like to understand why and if it is really needed.

The reason this is needed is that in an fhsenv (such as the envision build environment), auto-patchelf will find the libraries in the fhsenv due to the binaries' rpaths, and won't modify the binaries that need the rpaths to point into the nix store.